packages/worker/src/worker.ts

```diff
@@ -505,10 +506,13 @@ export class Worker {
   ): Promise<void> {
     const validated = this.validateHistory(history);
     try {
-      await this.runReplayHistories(
-        { ...options, failFast: true }, // Always failFast in single replay
-        [{ history: validated, workflowId: workflowId ?? 'fake' }]
-      );
+      for await (const result of this.runReplayHistories(options, [
```

Comment:
I'd make some `first` or `single` function in iterator-utils.ts... That would make it clear that we are expecting only one result here.

Reply:
Hmm.. sure